lib/publish.js

```diff
@@ -77,10 +77,12 @@ module.exports = async (pluginConfig, context) => {
       description: notes && notes.trim() ? notes : gitTag,
       milestones,
       assets: {
-        links: assetsList.map(({label, alt, url}) => {
+        links: assetsList.map(({label, alt, url, type, filepath}) => {
           return {
             name: label || alt,
             url: urlJoin(gitlabUrl, repoId, url),
+            link_type: type || 'other',
```

Comment:
I can't remember off the top of my head if old versions of GitLab (i.e. before `link_type` was introduced) will accept POST requests that include fields it doesn't recognize 🤔 

If they _don't_, this will cause an error for older version of GitLab. But I'm about 90% sure `link_type` would just be ignored.

Regardless, I think it would be best to just pass `type` here:

```suggestion
            link_type: type
```

This has two advantages:

1. It defers to GitLab to determine what the default should be if no `link_type` is passed. No need to update this code if the default changes on GitLab's side
2. It passes `undefined` to the API if no `link_type` is configured, side stepping the issue mentioned above (I think - I haven't tested this)

What do you think?